mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,786 @@
+"""
+The ``mlflow.johnsnowlabs`` module provides an API for logging and loading Spark NLP and NLU models. This module
+exports the following flavors:
+
+The following Environment Variables must be present
+- `SECRET`: The secret for the John Snow Labs Enterprise NLP Library
+- `SPARK_NLP_LICENSE`: Your John Snow Labs Enterprise NLP License
+- `AWS_ACCESS_KEY_ID`: Your AWS Secret ID for accessing John Snow Labs Enterprise Models
+- `AWS_SECRET_ACCESS_KEY`: Your AWS Secret key for accessing John Snow Labs Enterprise Models
+
+
+You can set them like this in Python
+
+```python
+import os
+os.environ['SECRET'] = 'MY_SECRET'
+os.environ['AWS_ACCESS_KEY_ID'] = 'MY_AWS_ACCESS_KEY_ID'
+os.environ['AWS_SECRET_ACCESS_KEY'] = 'MY_AWS_SECRET_ACCESS_KEY'
+os.environ['SPARK_NLP_LICENSE'] = 'MY_SPARK_NLP_LICENSE'
+
+
+Johnsnowlabs (native) format
+    Allows models to be loaded as Spark Transformers for scoring in a Spark session.
+    Models with this flavor can be loaded as NluPipelines, with underlying Spark MLlib PipelineModel
+    This is the main flavor and is always produced.
+:py:mod:`mlflow.pyfunc`
+    Supports deployment outside of Spark by instantiating a SparkContext and reading
+    input data as a Spark DataFrame prior to scoring. Also supports deployment in Spark
+    as a Spark UDF. Models with this flavor can be loaded as Python functions
+    for performing inference. This flavor is always produced.
+:py:mod:`mlflow.mleap`
+    Enables high-performance deployment outside of Spark by leveraging MLeap's
+    custom dataframe and pipeline representations. Models with this flavor *cannot* be loaded
+    back as Python objects. Rather, they must be deserialized in Java using the
+    ``mlflow/java`` package. This flavor is produced only if you specify
+    MLeap-compatible arguments.
+"""
+import json
+import logging
+import os
+import posixpath
+import shutil
+import sys
+from pathlib import Path
+from urllib.request import urlopen
+
+import yaml
+from johnsnowlabs import nlp, settings
+from johnsnowlabs.auto_install.jsl_home import get_install_suite_from_jsl_home
+from johnsnowlabs.py_models.jsl_secrets import JslSecrets
```

Comment:
Can we import `johnsnowlabs` in functions?

Reply:
makes sense updated
resolved in https://github.com/mlflow/mlflow/pull/8556/commits/76d55dcdd647b6d183e933e3f7802995c95fc2fc